tests/Unleash.Tests/Communication/UnleashApiClient_Metrics_Backoff_Tests.cs

```diff
@@ -0,0 +1,251 @@
+
+        var apiClient = new UnleashApiClient(
+            httpClient,
+            A.Fake<IJsonSerializer>(),
+            A.Fake<UnleashApiClientRequestHeaders>(),
+            new EventCallbackConfig()
+        );
+
+        // Act
+        var result = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+        var result2 = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+        
+        // Assert
+        messageHandler.CallCount.Should().Be(2);
+    }
+
+    [Test]
+    public void Backoff_Caps_Out_At_10()
+    {
+        // Arrange
+        var messageHandler = new CountingConfigurableHttpMessageHandler
+        (
+            new List<HttpResponseMessage>
+            {
+                TooManyRequests,
+                TooManyRequests,
+                InternalServerError,
+                TooManyRequests,
+                TooManyRequests,
+                InternalServerError,
+                InternalServerError,
+                GatewayTimeout,
+                BadGateway,
+                ServiceUnavailable,
+                Forbidden,
+                Unauthorized,
+                Ok
+            }
+        );
+        var httpClient = new HttpClient(messageHandler);
+        httpClient.BaseAddress = new Uri("http://localhost:8080/");
+
+        var apiClient = new UnleashApiClient(
+            httpClient,
+            A.Fake<IJsonSerializer>(),
+            A.Fake<UnleashApiClientRequestHeaders>(),
+            new EventCallbackConfig()
+        );
+
+        // Act
+        var result = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+        
+        // Assert
+        messageHandler.CallCount.Should().Be(1);
+    }
+
+    [Test]
+    public void Should_Skip_One_After_First_429()
+    {
+        // Arrange
+        var messageHandler = new CountingConfigurableHttpMessageHandler
+        (
+            new List<HttpResponseMessage>
+            {
+                TooManyRequests,
+                Ok
+            }
+        );
+        var apiClient = GetClient(messageHandler);
+
+        // Act 1
+        var result = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+
+        // Assert 1
+        messageHandler.CallCount.Should().Be(1);
+
+        // Act 2
+        var result2 = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+        var result3 = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+
+        // Assert 2
+        messageHandler.CallCount.Should().Be(2);
+    }
+
+    [Test]
+    public void Access_Error_Responses_Goes_Straight_To_Ten()
```

Comment:
Just looking at the test name, I don't see a check for responses == 10? Is that intentional? Am I reading the test wrong?

Reply:
Yeah can't check it directly, so there's a loop of 10 in there along with a check that call count hasn't increased as a result of that loop - indicating it went straight to skipping 10, and then the additional invocation that should be let through the skip check and increase call count by 1